models/pyannote.py

```diff
@@ -0,0 +1,53 @@
+# pip install pyannote.metrics
+# pip install pyannote.pipeline
+# pip install pescador
+# pip install optuna
+# pip install filelock
+# pip install git+https://github.com/pyannote/pyannote-audio
+# pip install sortedcontainers simplejson typing_extensions pyannote.database pyannote.metrics
+# pip install pyannote.core --no-dependencies
+
+import shapes
+import torch
+import numpy as np
+
+
+class PyannoteDiarizationModel:
+	def __init__(self, vad, vad_sensitivity: float, sample_rate: int):
+		assert sample_rate == 16_000, 'Only 16 kHz sample rate supported'
+		self.vad = vad
+		self.vad_sensitivity = vad_sensitivity
+		self.sample_rate = sample_rate
+		self.model = torch.hub.load('pyannote/pyannote-audio', 'dia', batch_size = 1)
+
+	@property
+	def input_dtype(self):
+		return self.vad.input_dtype
+
+	def get_silence_mask(self, signal: shapes.BT):
+		if self.vad.input_type == np.array:
+			device = signal.device
+			signal = signal.cpu().numpy()
+			silence_mask = torch.as_tensor(self.vad.detect(signal)[0], dtype = torch.bool, device = device)
+		else:
+			silence_mask = self.vad.detect(signal)[0]
+		return silence_mask
+
+	def get_speaker_mask(self, signal: shapes.BT, sample_rate: int):
```

Comment:
let's move these `get_speaker_mask` methods to just be `forward` and upgrade the class to derive from nn.Module?

Reply:
Motivation? What if we will use model not based on pytorch?